### PR TITLE
Change storage version for resources in api.astarte-platform.org

### PR DESCRIPTION
**Pull request by matt-mazzucato**
- Resources in the api.astarte-platform.org group are now stored in the v1alpha2 version
- Add a deprecation notice for Astarte's and Flow's v1alpha1 api versions
- Add a deprecation notice for AstarteVoyagerIngress (the API will no longer be available starting from Astarte Operator v1.2+)

### CODE REVIEW

**Review by Annopaolo — CHANGES_REQUESTED** [review]

File: lib/migrate/migrate.go [hunk 1]
```diff
@@ -35,7 +35,7 @@ import (
 	"k8s.io/apimachinery/pkg/types"
 	"sigs.k8s.io/controller-runtime/pkg/client"
 
-	"github.com/astarte-platform/astarte-kubernetes-operator/apis/api/v1alpha1"
+	"github.com/astarte-platform/astarte-kubernetes-operator/apis/api/v1alpha2"
 )
 
 // The purpose of this package is solely the migration of the old operator's flaky CR to the shiny new, stable,
```
[the remark this answers is not in the document]
Comment (matt-mazzucato) [reply lib/migrate/migrate.go:41]: Let's tackle this in a dedicated PR

File: apis/api/v1alpha1/conversion.go [hunk 2]
```diff
@@ -18,7 +18,77 @@
...
+
+// ConvertTo converts this Astarte to the Hub version (v1alpha2).
+func (src *Astarte) ConvertTo(dstRaw conversion.Hub) error {
+	dst := dstRaw.(*v1alpha2.Astarte)
+	err := Convert_v1alpha1_Astarte_To_v1alpha2_Astarte(src, dst, nil)
+	if err != nil {
+		return err
+	}
+
+	return nil
```
Comment (Annopaolo) [inline apis/api/v1alpha1/conversion.go:35]: Could we shorten those lines like this? If there is no error, `nil` will be returned anyway.
```go
return Convert_v1alpha1_Astarte_To_v1alpha2_Astarte(src, dst, nil)
```
The same applies to all other conversion functions in this file.

**Review by Annopaolo — CHANGES_REQUESTED** [review]

**Review by Annopaolo — APPROVED** [review]

**Review by rbino — APPROVED** [review]

File: CHANGELOG.md [hunk 3]
```diff
@@ -12,6 +12,10 @@ and this project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.
 - Remove common types. Automate CRD conversion functions generation.
 - Reconcile AstarteDefaultIngress when the Astarte CR changes.
 - Upgrade OperatorSDK to v1.21.0.
+- Add deprecation notice for v1alpha1 resources (i.e. Astarte and AstarteFlow).
```
Comment (Annopaolo) [inline CHANGELOG.md:15]: Add also here `(will be unavailable starting from Astarte Operator v1.2+).`

File: CHANGELOG.md [hunk 4]
```diff
@@ -12,6 +12,9 @@ and this project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.
 - Remove common types. Automate CRD conversion functions generation.
 - Reconcile AstarteDefaultIngress when the Astarte CR changes.
 - Upgrade OperatorSDK to v1.21.0.
+- Add deprecation notice for v1alpha1 resources in the api.astarte-platform.org group: i.e. Astarte,
+  AstarteFlow and AstarteVoyagerIngresses will be unavailable starting from Astarte Operator v1.2+).
```
Comment (Annopaolo) [inline CHANGELOG.md:16]: Remove leftover `)`

**Review by Annopaolo — APPROVED** [review]

**Review by rbino — APPROVED** [review]

**Pull request merged**